tensornetwork/backends/numpy/numpy_backend.py

```diff
@@ -283,7 +283,146 @@ def matvec(vector):
     if dtype:
       eta = eta.astype(dtype)
       U = U.astype(dtype)
-    return eta, [np.reshape(U[:, n], shape) for n in range(numeig)]
+    evs = list(eta)
+    eVs = [np.reshape(U[:, n], shape) for n in range(numeig)]
+    return evs, eVs
+
+  def gmres(self,
```

Comment:
should we also add lgmres? I found lgmres to be more stable and also faster than gmres in my applications, though it's been a few years since I have last tested this.

Reply:
We should (it is, indeed, just better) but perhaps not yet, since I haven't written it in Jax.

Reply:
We also should add an iterative symmetric solver like CGS.

Reply:
let's submit two issues then